src/lib/error/exceeds-limit-error.ts

```diff
@@ -9,3 +11,12 @@ export class ExceedsLimitError extends GenericUnleashError {
         });
     }
 }
+
+export const throwExceedsLimitError = (
```

Comment:
I'd put eventBus as the first param since it's a complex dependency and not data. As a rule of thumb I prefer complex dependencies first and data second.
```
class Class {
   constructure(dep) {}
  method(data) {}
}```
becomes
```fn(dep, data) {}```
If you curry this function you can bake deps in and you're left with the data.

There's a great talk on this topic: https://www.youtube.com/watch?v=m3svKOdZijA